pkg/ddl/partition.go

```diff
@@ -1578,12 +1580,21 @@ func checkResultOK(ok bool) error {
 }
 
 // checkPartitionFuncType checks partition function return type.
-func checkPartitionFuncType(ctx sessionctx.Context, expr ast.ExprNode, tblInfo *model.TableInfo) error {
+func checkPartitionFuncType(ctx sessionctx.Context, expr ast.ExprNode, dbName model.CIStr, tblInfo *model.TableInfo) error {
 	if expr == nil {
 		return nil
 	}
 
-	e, err := expression.RewriteSimpleExprWithTableInfo(ctx, tblInfo, expr, false)
```

Comment:
Could we modify `RewriteSimpleExprWithTableInfo` directly? `RewriteSimpleExprWithTableInfo` is used in one other place besides here, can that place be changed in the same way? This will remove the duplicate code here.